pkg/gofr/datasource/pubsub/eventhub/eventhub.go

```diff
@@ -0,0 +1,359 @@
+	"github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs/checkpoints"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/container"
+
+	"gofr.dev/pkg/gofr/datasource"
+	"gofr.dev/pkg/gofr/datasource/pubsub"
+)
+
+// code reference from https://learn.microsoft.com/en-us/azure/event-hubs/event-hubs-go-get-started-send
+
+type Config struct {
+	ConnectionString          string
+	ContainerConnectionString string
+	StorageServiceURL         string
+	StorageContainerName      string
+	EventhubName              string
+	// if not provided it will read from the $Default consumergroup.
+	ConsumerGroup string
+	// following configs are for advance setup of the eventhub.
+	StorageOptions   *container.ClientOptions
+	BlobStoreOptions *checkpoints.BlobStoreOptions
+	ConsumerOptions  *azeventhubs.ConsumerClientOptions
+	ProducerOptions  *azeventhubs.ProducerClientOptions
+}
+
+type Client struct {
+	producer *azeventhubs.ProducerClient
+	consumer *azeventhubs.ConsumerClient
+	// we are using processor such that to keep consuming the events from all the different partitions.
+	processor *azeventhubs.Processor
+	// checkpoint is being called while committing the event received from the event.
+	checkPoint *checkpoints.BlobStore
+	// processorCtx is being stored such that to gracefully shutting down the application.
+	processorCtx context.CancelFunc
+	cfg          Config
+	logger       Logger
+	metrics      Metrics
+	tracer       trace.Tracer
+}
+
+// New Creates the client for Eventhub
+func New(cfg Config) *Client {
+	return &Client{
+		cfg: cfg,
+	}
+}
+
+func (c *Client) validConfigs(cfg Config) bool {
+	ok := true
+
+	if cfg.EventhubName == "" {
+		ok = false
+
+		c.logger.Error("eventhubName cannot be an empty")
+	}
+
+	if cfg.ConnectionString == "" {
+		ok = false
+
+		c.logger.Error("connectionString cannot be an empty")
+	}
+
+	if cfg.StorageServiceURL == "" {
+		ok = false
+
+		c.logger.Error("storageServiceURL cannot be an empty")
+	}
+
+	if cfg.StorageContainerName == "" {
+		ok = false
+
+		c.logger.Error("storageContainerName cannot be an empty")
+	}
+
+	if cfg.ContainerConnectionString == "" {
+		ok = false
+
+		c.logger.Error("containerConnectionString cannot be an empty")
+	}
+
+	if cfg.ConsumerGroup == "" {
+		cfg.ConsumerGroup = azeventhubs.DefaultConsumerGroup
+	}
+
+	return ok
+
+}
+
+// UseLogger sets the logger for the Cassandra client.
+func (c *Client) UseLogger(logger any) {
+	if l, ok := logger.(Logger); ok {
+		c.logger = l
+	}
+}
+
+// UseMetrics sets the metrics for the Cassandra client.
+func (c *Client) UseMetrics(metrics any) {
+	if m, ok := metrics.(Metrics); ok {
+		c.metrics = m
+	}
+}
+
+// UseTracer sets the tracer for the MongoDB client.
+func (c *Client) UseTracer(tracer any) {
+	if t, ok := tracer.(trace.Tracer); ok {
+		c.tracer = t
+	}
+}
+
+// Connect establishes a connection to Cassandra and registers metrics using the provided configuration when the client was Created.
```

Comment:
Correct me if I am wrong.. but I think we are missing metrics registration.
- If we are registering them at any other place then comment needs to be fixed. OR
- If registration is missing, we would need to do that.

Reply:
Pubsub metrics are being registered in container - and we are using the same metrics so we need not to register the metrics here.
Will add the comments for the same